iceoryx_posh/include/iceoryx_posh/popo/wait_set.hpp

```diff
@@ -26,129 +31,96 @@ class Condition;
-    /// @return ConditionVector vector of condition pointers that have become
-    /// fulfilled
-    ConditionVector wait() noexcept;
-
-    /// @brief Is a given condition attached to this WaitSet
-    /// @return true if the condition is attached otherwise false
-    bool isConditionAttached(const Condition& condition) noexcept;
+    /// @brief Acquires a trigger from the waitset  The trigger is then attached to a class, the origin,
+    ///        which triggers the Trigger if a specific event happens. The class must then signal the
+    ///        trigger that it was triggered via the triggerCallback. If the WaitSet goes out of scope
+    ///        before the class does it calls the invalidationCallback to invalidate the Trigger inside
+    ///        of the class
+    ///        You cannot acquire an already logical equal acquired trigger. This means if you acquire a trigger
+    ///        twice with the same: origin, triggerCallback and triggerId this method will return
+    ///        TRIGGER_ALREADY_ACQUIRED
+    /// @param[in] origin the pointer to the class which will attach the trigger
+    /// @param[in] triggerCallback a method from the class which will signal the Trigger that it was triggered
+    /// @param[in] invalidationCallback callback which will be called in the destructor of the waitset, important when
+    /// the waitset goes out of scope before the origin does.
+    /// @param[in] triggerId an arbitrary id to identify the trigger later when a list of triggers is returned via wait
+    /// or timedWait
+    /// @param[in] callback a callback which is attached to the trigger and can be later called when the trigger will be
+    ///            returned via wait or timedWait
+    /// @return returns the newly created trigger if the WaitSet has space left otherwise it returns
+    /// WaitSetError::TRIGGER_VECTOR_OVERFLOW
+    template <typename T>
+    cxx::expected<Trigger, WaitSetError>
+    acquireTrigger(T* const origin,
+                   const cxx::ConstMethodCallback<bool>& triggerCallback,
+                   const cxx::MethodCallback<void, const Trigger&>& invalidationCallback,
+                   const uint64_t triggerId = Trigger::INVALID_TRIGGER_ID,
+                   const Trigger::Callback<T> callback = nullptr) noexcept;
+
+    /// @brief Blocking wait with time limit till one or more of the triggers are triggered
+    /// @param[in] timeout How long shall we waite for a trigger
+    /// @return TriggerStateVector of TriggerStates that have been triggered
+    TriggerStateVector timedWait(const units::Duration timeout) noexcept;
+
+    /// @brief Blocking wait till one or more of the triggers are triggered
+    /// @return TriggerStateVector of TriggerStates that have been triggered
+    TriggerStateVector wait() noexcept;
+
+    /// @brief Returns the amount of stored Trigger inside of the WaitSet
+    uint64_t size() const noexcept;
+
+    /// @brief Returns to capacity
```

Comment:
which capacity. The argument that the STL container also have a capacity method is not valid for me. A waitset is not a classical conatiner but has some containers inside

Reply:
renamed to `triggerCapacity` and the description is more detailled